src/main/scala/sbt/errorssummary/Plugin.scala

```diff
@@ -34,10 +34,20 @@ object Plugin extends AutoPlugin {
 
   private val reporterSettings = Seq(
     compilerReporter in compile := {
-      val logger  = streams.value.log
-      val baseDir = sys.props("user.dir") + File.separator
-      val spms    = Defaults.foldMappers(sourcePositionMappers.value)
-      val config  = (reporterConfig in compile).value
+      val logger     = streams.value.log
+      val baseDir    = sys.props("user.dir") + File.separator
+      val spms       = Defaults.foldMappers(sourcePositionMappers.value)
+      val baseConfig = (reporterConfig in compile).value
+
+      // When run in intellij, Emacs or when `sbti.errorssummary.full.paths = true`,
+      // don't shorten paths.
+      val forceFullPaths =
+        sys.props.contains("idea.runid") ||
```

Comment:
`idea.runid` should work, but I recommend checking `idea.managed=true`, since `runid` is only set in some contexts.